packages/webapp/src/components/Animals/AddAnimalsFormCard/AddAnimalsFormCard.tsx

```diff
@@ -78,45 +105,63 @@ export default function AddAnimalsFormCard({
         {showRemoveButton && <SmallButton variant="remove" onClick={onRemoveButtonClick} />}
       </div>
       <AnimalTypeSelect
-        name={`${namePrefix}.${FIELD_NAMES.TYPE}`}
+        name={`${namePrefix}${BasicsFields.TYPE}`}
         control={control}
         typeOptions={typeOptions}
-        onTypeChange={onTypeChange}
+        onTypeChange={(option) => {
+          trigger(`${namePrefix}${BasicsFields.TYPE}`);
+          onTypeChange?.(option);
+        }}
+        error={get(errors, `${namePrefix}${BasicsFields.TYPE}`)}
       />
       <AnimalBreedSelect
         breedSelectRef={breedSelectRef}
-        name={`${namePrefix}.${FIELD_NAMES.BREED}`}
+        name={`${namePrefix}${BasicsFields.BREED}`}
         control={control}
         breedOptions={filteredBreeds}
         isTypeSelected={!!watchAnimalType}
       />
 
       <div className={styles.countAndSexDetailsWrapper}>
         <NumberInput
-          name={`${namePrefix}.${FIELD_NAMES.COUNT}`}
+          name={`${namePrefix}${BasicsFields.COUNT}`}
           control={control}
-          defaultValue={0}
           label={t('common:COUNT')}
           className={styles.countInput}
           allowDecimal={false}
           showStepper
+          rules={{
+            required: {
+              value: true,
+              message: t('common:REQUIRED'),
+            },
+            min: hookFormMinValidation(1),
+          }}
+          onChange={() => trigger(`${namePrefix}${BasicsFields.COUNT}`)}
         />
         <Controller
-          name={`${namePrefix}.${FIELD_NAMES.SEX_DETAILS}`}
+          name={`${namePrefix}${BasicsFields.SEX_DETAILS}`}
           control={control}
-          render={({ field }) => (
+          rules={{
+            validate: (details: SexDetailsType) => {
+              if (!details) return true;
+              const total = details.reduce((prevCount, { count }) => prevCount + count, 0);
+              return total <= watchAnimalCount || 'Invalid sexDetails for count';
+            },
```

Comment:
I don't think it's worth spending the time, but having the same logic in validate and `SexDetails` doesn't feel right 😓

Reply:
I think I agreed with you on both counts here 😄 I initially tried passing the error as a prop after I added this validation, but when I saw that the message didn't triggering in the same way, I left it alone rather than looking any more deeply into how SexDetails was using `error`... it didn't seem worth the investigation at the time.

I think there is still a bit of room for refactor here though (like also maybe that extracted count + sexDetails component!) when we continue with the components in `readonly`/`edit`!